server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java

```diff
@@ -275,6 +277,87 @@
     assertThat(s3).doesNotContain("secret17");
   }
 
+  @Test public void secretVersionListing_notFound() throws Exception {
+    Request put = clientRequest("/automation/v2/secrets/non-existent/versions/0-0").build();
+    Response httpResponse = mutualSslClient.newCall(put).execute();
+    assertThat(httpResponse.code()).isEqualTo(404);
+  }
+
+  @Test public void secretVersionListing_success() throws Exception {
+    int totalVersions = 6;
+    int sleepInterval = 1000; // Delay so secrets have different creation timestamps
```

Comment:
Unfortunately, the secret's update and creation timestamps aren't affected by any of the requests to create or update a secret--it looks like Keywhiz automatically sets the timestamps.  I agree it'd be much better to set them manually, though; is there a workaround I could use?  (I could write and implement a new request to reset secret creation and update times, but I'm not sure whether that's worth it.)